### PR TITLE
[SPARK-41247][BUILD] Unify the Protobuf versions in Spark connect and Protobuf connector

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Unify the Protobuf versions in Spark connect and Protobuf connector. 

### Why are the changes needed?

The Protobuf dependencies will have the same behavior. And future upgrades will be easier.

### Does this PR introduce _any_ user-facing change?

No
### How was this patch tested?

GA tests

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -118,7 +118,10 @@
     <log4j.version>2.19.0</log4j.version>
     <!-- make sure to update IsolatedClientLoader whenever this version is changed -->
     <hadoop.version>3.3.4</hadoop.version>
-    <protobuf.version>2.5.0</protobuf.version>
+    <!-- Protobuf version for building with Hadoop/Yarn dependencies -->
+    <protobuf.hadoopDependency.version>2.5.0</protobuf.hadoopDependency.version>
+    <!-- Actual Protobuf version in Spark modules like connect, protobuf connector, etc. -->
```
Comment (cloud-fan) [inline pom.xml:123]:
```suggestion
    <!-- Actual Protobuf version in Spark modules like Spark Connect, protobuf connector, etc. -->
```

**Comment by cloud-fan** [comment]
shall we also shade protobuf in Spark Core in this PR?

**Comment by gengliangwang** [comment]
> shall we also shade protobuf in Spark Core in this PR? Then the proto version is unified in all the places.

In this PR there is no actual usage. I would prefer to shade it in https://github.com/apache/spark/pull/38779

**Review by cloud-fan — APPROVED** [review]

**Review by zhengruifeng — APPROVED** [review]
lgtm, also cc @HyukjinKwon @grundprinzip

**Review by grundprinzip — APPROVED** [review]

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

**Review by HyukjinKwon — APPROVED** [review]

**Comment by gengliangwang** [comment]
Merging to master

**Comment by amaliujia** [comment]
Now the protobuf version is unified for the scala side (or a few module of the scala side). 

But it is still different from the Python side (at least Connect Python client): https://github.com/apache/spark/pull/38774

cc @grundprinzip @cloud-fan @HyukjinKwon

**Pull request closed without merging**